impl/src/main/java/com/okta/sdk/impl/ds/DefaultResourceFactory.java

```diff
@@ -47,10 +47,21 @@ public DefaultResourceFactory(InternalDataStore dataStore) {
             throw new NullPointerException("Resource class cannot be null.");
         }
 
-        Class<T> implClass = getImplementationClass(clazz);
+        Class<T> classToResolve = clazz;
 
         Object[] ctorArgs = createConstructorArgs(constructorArgs);
 
+        if (ctorArgs.length >= 2 && ctorArgs[1] instanceof Map) {
```

Comment:
Check to see if there are multiple implementations.
With the Factors API, there are multiple types of factors, and the data map must be inspected before figuring out the correct interface